test/test_example.py

```diff
@@ -89,13 +89,28 @@ def test_flaky_function(param=[]):
     assert param_length == 1
 
 
+@flaky
 @genty
-class ExampleFlakyTestsWithUnicodeTestNames(ExampleFlakyTests):
-    @genty_dataset('ascii name', 'ńőń ȁŝćȉȉ ŝƭȕƒƒ')
-    def test_non_flaky_thing(self, message):
+class ExampleFlakyTestsWithTrickyTestNames(TestCase):
+    _threshold = -1
+
+    _unicode_string = 'Plain Hello'
+    _byte_string = str(_unicode_string)
+    _unicode_string_non_ascii = 'ńőń ȁŝćȉȉ ŝƭȕƒƒ'
+    _byte_string_non_ascii = _unicode_string_non_ascii.encode('utf-8')
+    _byte_string_windows_encoded = 'Hèllö'.encode('windows-1252')
+
+    @genty_dataset(
+        _unicode_string,
+        _byte_string,
+        _unicode_string_non_ascii,
+        _byte_string_non_ascii,
+        _byte_string_windows_encoded,
+    )
+    def test_tricky_strings_in_test_method_name(self, message):
         # pylint:disable=unused-argument
         self._threshold += 1
```

Comment:
`_threshold` should be set to -1 in `setUp()`, otherwise all the tests but the first will be no-ops.